pkg/secretlock/local/local_secret_lock.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package local
+
+import (
+	"crypto/cipher"
+	"encoding/base64"
+	"io"
+	"os"
+
+	"github.com/google/tink/go/subtle/random"
+	"golang.org/x/crypto/chacha20poly1305"
+
+	"github.com/hyperledger/aries-framework-go/pkg/secretlock"
+)
+
+// Lock is a secret lock service responsible for encrypting keys using a master key stored as environment variable
+type Lock struct {
+	aead cipher.AEAD
+}
+
+// NewService creates a new instance of a local secret lock service for a master key in path
+func NewService(path string) (secretlock.Service, error) {
```

Comment:
- should have more generic base version that uses io.writer.

- need to allow for optional passphrase option to protect key files. use secret expansion algorithm.
- might be sensible to chain secret locks together... have a passphrase instance nested on top of io.writer instance.

Reply:
it's probably good for a passphrase implementation, we're using pure crypto here which requires a real crypto key

again, secret locks are not meant to be exposed to the user, it's an internal service (used by KMS primarily)

Reply:
@Baha-sk there are other places a key can come from other than the disk. Better to use io.Writer as the base - I don't want to deal with each variation individually.

Reply:
@Baha-sk I think you also missed the chaining comment. There are deployments where there is an encrypted key in *some* storage and that key needs to be decrypted by a passphrase. The decrypted secret is then used.

Reply:
I'll add a variation that takes an io.Reader as option

Reply:
added